# Analysis of Airbnb Marketplace in Vancouver

## Objective

Vancouver is an expensive place to live. Housing costs vary a lot from one area to the next, so it takes time to learn about the different neighbourhoods before you settle in.

Airbnb is a desirable option for newcomers for the following reasons: 

Preferably before landing, newcomers may want to set up temporary accommodation for when they immediately arrive in the city. The online marketplace is accessible worldwide. 

Moreover, the prices of the listing on Airbnb are generally more economical than hotels for the short-term renting. 

In addition, the temporary accommodations offer newcomers more flexibility compared with fixed term tenancies.

Currently there are a few analyses which provide a reference and insights into the market. But they focus solely on the data itself. The analysis here is part of the **"Know Before You Go"** project and will be fed to another Machine Learning system, along with many other attributes in other datasets. The downstream system will determine the correlation among the data and offer target audience reference solutions.

## Questions

The quetions are categozied into three major features in terms of the assumptions for newcomers who care about **locations** and **prices**.

**Locations**:
1. How listings are distributed in Vancouver and in which areas listings are distributed densely?
2. How are property types dispersed throughout the city?

**Prices**:
3. Where are the most expensive/economical neighbourhoods to rent an Airbnb?
4. Which listings are undervalued – more beneficial to necomers?

**Others**:
5. How many houses/apartments are being rented out more frequently for the short-term purpose? 
6. Any impact from cancellation policies?


## Exploratory Data Analysis (EDA)

In [235]:
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"
import pandas as pd
import numpy as np

airbnb = pd.read_csv("listings_09Nov2019.csv")
airbnb.head()

id                         listing_url       scrape_id last_scraped  \
0  10080  https://www.airbnb.com/rooms/10080  20191109094845   2019-11-09   
1  13188  https://www.airbnb.com/rooms/13188  20191109094845   2019-11-09   
2  13357  https://www.airbnb.com/rooms/13357  20191109094845   2019-11-09   
3  13490  https://www.airbnb.com/rooms/13490  20191109094845   2019-11-09   
4  14267  https://www.airbnb.com/rooms/14267  20191109094845   2019-11-09   

                                                name  \
0                     D1 -  Million Dollar View 2 BR   
1                  Garden level studio in ideal loc.   
2  ! Wow! 2bed 2bath 1bed den Harbour View Apartm...   
3                       Vancouver's best kept secret   
4                                  EcoLoft Vancouver   

                                             summary  \
0  Stunning two bedroom, two bathroom apartment. ...   
1  Garden level studio suite with garden patio - ...   
2  Very spacious and comfortable with very well k...   
3  This apartment rents for one month blocks of t...   
4  The Ecoloft is located in the lovely, family r...   

                                               space  \
0  Bed setup: 2 x queen, option to add up to 2 tw...   
1  Very Close (3min walk) to Nat Bailey baseball ...   
2  Mountains and harbour view 2 bedroom,2 bath,1 ...   
3  Vancouver city central, 700 sq.ft., main floor...   
4  West Coast Modern Laneway House Loft: We call ...   

                                         description experiences_offered  \
0  Stunning two bedroom, two bathroom apartment. ...                none   
1  Garden level studio suite with garden patio - ...                none   
2  Very spacious and comfortable with very well k...                none   
3  This apartment rents for one month blocks of t...                none   
4  The Ecoloft is located in the lovely, family r...                none   

                               neighborhood_overview  \
0                                                NaN   
1  The uber hip Main street area is a short walk ...   
2  Amanzing bibrant professional neighbourhood. C...   
3  In the heart of Vancouver, this apartment has ...   
4  We live in the centre of the city of Vancouver...   

                                               notes  \
0  1. CHECK-IN TIME IS AFTER 3PM PST AND CHECK-OU...   
1  Parking is free on the streets to the east and...   
2  Some strata bylaws that need to follow and res...   
3  This apartment is designed for adult guests on...   
4  Our family includes a friendly German Shepard ...   

                                             transit  \
0                                                NaN   
1  Check out (Website hidden by Airbnb)  To see t...   
2                                                NaN   
3  This apartment could not be more central. It i...   
4  We are within a few blocks of three major arte...   

                                              access  \
0  Enjoy our amenities which include gym, pool, h...   
1                                                NaN   
2  The entire apartment . Except no included the ...   
3  Guests have full access to the beautiful back ...   
4  Guests have the entrance, staircase and second...   

                                         interaction  \
0                                                NaN   
1  We live in the house in which the suite is loc...   
2  I'm personally. Do that wellcome check in and ...   
3  We live on the second floor. The rental apartm...   
4  We see our guests coming and going if we are o...   

                                         house_rules  thumbnail_url  \
0  1. CHECK-IN TIME IS AFTER 3 PM PST AND CHECK-O...            NaN   
1   Limited noise after 11:00pm. Strictly NO SMOKING            NaN   
2  No parties,no smoking,no pets. Damages deposit...            NaN   
3  We offer a home for a single person or a coupl...            NaN   
4  We request absolutely no smoking

In [236]:
airbnb.shape
airbnb.info()
airbnb.columns.tolist()

(6181, 106)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6181 entries, 0 to 6180
Columns: 106 entries, id to reviews_per_month
dtypes: float64(23), int64(21), object(62)
memory usage: 5.0+ MB


['id',
 'listing_url',
 'scrape_id',
 'last_scraped',
 'name',
 'summary',
 'space',
 'description',
 'experiences_offered',
 'neighborhood_overview',
 'notes',
 'transit',
 'access',
 'interaction',
 'house_rules',
 'thumbnail_url',
 'medium_url',
 'picture_url',
 'xl_picture_url',
 'host_id',
 'host_url',
 'host_name',
 'host_since',
 'host_location',
 'host_about',
 'host_response_time',
 'host_response_rate',
 'host_acceptance_rate',
 'host_is_superhost',
 'host_thumbnail_url',
 'host_picture_url',
 'host_neighbourhood',
 'host_listings_count',
 'host_total_listings_count',
 'host_verifications',
 'host_has_profile_pic',
 'host_identity_verified',
 'street',
 'neighbourhood',
 'neighbourhood_cleansed',
 'neighbourhood_group_cleansed',
 'city',
 'state',
 'zipcode',
 'market',
 'smart_location',
 'country_code',
 'country',
 'latitude',
 'longitude',
 'is_location_exact',
 'property_type',
 'room_type',
 'accommodates',
 'bathrooms',
 'bedrooms',
 'beds',
 'bed_type',
 'amenities',


### 1. Numerical Attributes

There are 106 attributes in the dataset. Before selecting the numerical attributes, I need to remove the unnecessary symbols in the data, to be specific, in the columns of `host_response_rate`, `price`, `weekly_price`,	`monthly_price`, `security_deposit`,	`cleaning_fee`,	`extra_people`.

In [237]:
airbnb[['price', 'weekly_price', 'monthly_price', 'security_deposit', 'cleaning_fee','extra_people']] =\
    airbnb[['price', 'weekly_price', 'monthly_price', 'security_deposit', 'cleaning_fee','extra_people']].apply(lambda x: x.str.replace('$', ''))

airbnb[['price', 'weekly_price', 'monthly_price', 'security_deposit', 'cleaning_fee','extra_people']] =\
    airbnb[['price', 'weekly_price', 'monthly_price', 'security_deposit', 'cleaning_fee','extra_people']].apply(lambda x: x.str.replace(',', ''))

airbnb[['price', 'weekly_price', 'monthly_price', 'security_deposit', 'cleaning_fee','extra_people']] =\
    airbnb[['price', 'weekly_price', 'monthly_price', 'security_deposit', 'cleaning_fee','extra_people']].astype(float)

airbnb['host_response_rate'] = airbnb['host_response_rate'].str.replace('%', '').astype(float)
airbnb['host_response_rate'] = airbnb['host_response_rate']/100

In [238]:
type(airbnb['neighbourhood_group_cleansed'][1])

airbnb['neighbourhood_group_cleansed'] = airbnb['neighbourhood_group_cleansed'].astype(str)

numpy.float64

In [239]:
airbnb_num = airbnb.select_dtypes(include=['float64', 'int64'])

# show all the selected columns
pd.set_option('display.max_columns', None)
airbnb_num.describe()

id     scrape_id  thumbnail_url  medium_url  xl_picture_url  \
count  6.181000e+03  6.181000e+03            0.0         0.0             0.0   
mean   2.539230e+07  2.019111e+13            NaN         NaN             NaN   
std    1.159556e+07  1.918124e+00            NaN         NaN             NaN   
min    1.008000e+04  2.019111e+13            NaN         NaN             NaN   
25%    1.734767e+07  2.019111e+13            NaN         NaN             NaN   
50%    2.783392e+07  2.019111e+13            NaN         NaN             NaN   
75%    3.557718e+07  2.019111e+13            NaN         NaN             NaN   
max    3.997596e+07  2.019111e+13            NaN         NaN             NaN   

            host_id  host_response_rate  host_acceptance_rate  \
count  6.181000e+03         5400.000000                   0.0   
mean   1.050765e+08            0.969591                   NaN   
std    9.996218e+07            0.115652                   NaN   
min    6.033000e+03            0.000000                   NaN   
25%    1.553542e+07            1.000000                   NaN   
50%    6.595935e+07            1.000000                   NaN   
75%    1.937396e+08            1.000000                   NaN   
max    3.079859e+08            1.000000                   NaN   

       host_listings_count  host_total_listings_count     latitude  \
count          6175.000000                6175.000000  6181.000000   
mean              4.147368                   4.147368    49.260711   
std               9.639047                   9.639047     0.021476   
min               0.000000                   0.000000    49.202900   
25%               1.000000                   1.000000    49.246900   
50%               1.000000                   1.000000    49.265960   
75%               3.000000                   3.000000    49.278640   
max             107.000000                 107.000000    49.293880   

         longitude  accommodates    bathrooms     bedrooms         beds  \
count  6181.000000   6181.000000  6179.000000  6179.000000  6177.000000   
mean   -123.111563      3.533732     1.333954     1.527755     1.874858   
std       0.039538      2.162202     0.659939     1.107816     1.264570   
min    -123.218790      1.000000     0.000000     0.000000     0.000000   
25%    -123.130380      2.000000     1.000000     1.000000     1.000000   
50%    -123.115100      3.000000     1.000000     1.000000     2.000000   
75%    -123.087580      4.000000     1.500000     2.000000     2.000000   
max    -123.017490     16.000000     8.500000    30.000000    15.000000   

        square_feet         price  weekly_price  monthly_price  \
count     75.000000   6181.000000    513.000000     553.000000   
mean     941.573333    180.013914    802.783626    2750.443038   
std     2780.241722    306.978613    499.633410    1756.025512   
min        0.000000     15.000000    150.000000     530.000000   
25%        0.000000     85.000000    459.000000    1650.000000   
50%       60.000000    129.000000    700.000000    2400.000000   
75%      850.000000    199.000000    980.000000    3495.000000   
max    21528.000000  13000.000000   3500.000000   12995.000000   

       security_deposit  cleaning_fee  guests_included  extra_people  \
count       5298.000000   5735.000000      6181.000000   6181.000000   
mean         414.979237     84.973845         1.972335     16.402200   
std          679.964019     65.007628         1.624164     27.879486   
min            0.000000      0.000000         1.000000      0.000000   
25%          125.000000     40.000000         1.000000      0.000000   
50%          250.000000     75.000000         1.000000     10.000000   
75%          500.000000    100.000000         2.000000     25.000000   
max         6709.000000    660.000000        16.000000    399.000000   

       minimum_nights  maximum_nights  minimum_minimum_nights  \
count     6181.000000     6181.000000             6181.000000   
mean        11.721243      67

In [ ]:
%matplotlib inline
import matplotlib.pyplot as plt

price = airbnb[['price', 'weekly_price', 'monthly_price', 
                'security_deposit', 'cleaning_fee', 'extra_people']]

price.hist(bins = 50, figsize = (15, 10))

array([[<matplotlib.axes._subplots.AxesSubplot object at 0x000001AFEB12EC88>,
      dtype=object)

In [ ]:
basic = airbnb[['accommodates', 'bathrooms', 'bedrooms', 'beds', 'square_feet', 'guests_included']]

basic.hist(bins = 10, figsize = (15, 10))

In [ ]:
availability = airbnb_num[['availability_30', 'availability_60', 'availability_90', 'availability_365']]

availability.hist(bins = 20, figsize = (15, 10))

In [ ]:
nights = airbnb_num[['minimum_nights', 'maximum_nights']]

nights.hist(bins = 50, figsize = (15, 5))

In [ ]:
reviews = airbnb_num[['number_of_reviews', 'review_scores_rating', 'review_scores_accuracy', 
                      'review_scores_cleanliness', 'review_scores_checkin', 'review_scores_communication', 
                      'review_scores_location', 'review_scores_value']]

reviews.hist(bins = 50, figsize = (20, 15))

From the above histograms we find there are some outliers. To be specific, the distributions of the `review score` related attributes are left skewed while the distributions of the `basic` attributes and price-related attributes are right skewed. For example, some value in (daily) `price` is more than $2000. 

As <a href="https://github.com/vcai01/know_before_you_go">the target audience</a> is clarified, it is less likely for newcomers to rent a short-term place accommodating more than the number of their household members. Also, it is unusual for newcomers to spend more than $1000 on short-term accommodation for only one night, so I subset the data and use the `airbnb_newcomers` hereafter.

In [ ]:
airbnb_newcomers = airbnb[(airbnb['accommodates'] <= 6) & (airbnb['price'] < 1000)]

len(airbnb_newcomers)

### 2.  Categorical Attributes



In [ ]:
from plotnine import *

airbnb_melted = pd.melt(airbnb_newcomers, id_vars = ['host_id'], value_vars = ['host_identity_verified', 'host_is_superhost']) 

airbnb_host_count = airbnb_melted.replace(np.NaN,'NA').groupby(['value', 'variable']).count().reset_index()
airbnb_host_count.rename(columns = {'host_id': 'Count'}, inplace = True)

airbnb_host_count

In [ ]:
ggplot(airbnb_host_count, aes(x='variable', y='Count', fill = 'value')) +\
    geom_bar(stat = 'sum', position = 'dodge') +\
    geom_text(aes(label = 'Count'), position = position_dodge(0.9), vjust = 'bottom')

### 3. Missing Values

In [ ]:
airbnb_newcomers_num = airbnb_newcomers.select_dtypes(include=['float64', 'int64'])

airbnb_newcomers_num.describe()

### 4. Visualizng Geographical Data

In [ ]:
airbnb_newcomers.plot(kind = 'scatter', x = 'longitude', y = 'latitude', alpha = 0.4,
                     c = 'price', cmap = plt.get_cmap('jet'), colorbar = True)

In [ ]:
airbnb_newcomers['longitude'].min()
airbnb_newcomers['longitude'].max()
airbnb_newcomers['latitude'].min()
airbnb_newcomers['latitude'].max()

In [ ]:
import matplotlib.image as mpimg

Vancouver_img=mpimg.imread('Vancouver_img.png')

airbnb_newcomers.plot(kind = 'scatter', x = 'longitude', y = 'latitude', alpha = 0.4,
                      figsize = (20, 10), c = 'price', cmap = plt.get_cmap('jet'), colorbar = True)

plt.title('Vancouver Airbnb Prices', fontsize = 16)

plt.imshow(Vancouver_img, extent=[-123.33, -122.902, 49.184, 49.328], alpha = 0.4,
           cmap=plt.get_cmap("jet"))

plt.ylabel("Latitude", fontsize=14)
plt.xlabel("Longitude", fontsize=14)

In [ ]:
corr_matrix = airbnb_newcomers.corr()

corr_matrix['price'].sort_values(ascending = False)

In [ ]:
from pandas.plotting import scatter_matrix

attributes = ['review_scores_rating', 'review_scores_accuracy', 'review_scores_value', 
              'price', 'host_response_rate']

scatter_matrix(airbnb_newcomers[attributes], figsize = (15, 8))